src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -494,9 +494,9 @@ private boolean addConnection()
          String password = config.getPassword();
 
          connection = (username == null) ? dataSource.getConnection() : dataSource.getConnection(username, password);
-         poolElf.setupConnection(connection, connectionTimeout);
+         PoolBagEntry pbe = poolElf.createBagEntry(connection, connectionTimeout, this);
 
-         connectionBag.add(new PoolBagEntry(connection, this));
+         connectionBag.add(pbe);
```

Comment:
true. but I am sure it is not that difficult to understand either having pbe defined right above the line.
this change removes juggling of method calls while creating bag entry with one one less method